### ISSUE TITLE
How to train model without end to end training?

### ISSUE THREAD

**Issue by Len-Li** [tjiiv-cprg/MonoRUn#1]
Hi, thanks for you sharing your work.

I found that all the configs are using end to end training. So how to train without backpropagatable pnp?
I try to throw out the pose_head, which result in worse result. How can I reproduce it as paper reported?

Thanks.

**Comment by Lakonik** [comment]
Hi! Thank you for the interest in our work.
End-to-end training with differentiable PnP is not included in this repository. The pose head is used for covariance calibration and IoU calculation. The configs are based on the _Fully Self-Supervised Reconstruction (Without Extra Supervision)_ and _LiDAR Supervision_ section of the paper, you can reproduce the results with the provided configs.

**Comment by Len-Li** [comment]
I got it. Thanks for your reply!

**Comment by kaixinbear** [comment]
> Hi! Thank you for the interest in our work.
> End-to-end training with differentiable PnP is not included in this repository. The pose head is used for covariance calibration and IoU calculation. The configs are based on the _Fully Self-Supervised Reconstruction (Without Extra Supervision)_ and _LiDAR Supervision_ section of the paper, you can reproduce the results with the provided configs.

Hi,  Lakonik~ Thansk for your great work. I wanna to know you two questions.
1. It seems that the code base you mean is not corresponding to the paper? 

> Leveraging the differentiable PnP, we apply smooth L1 loss on the Euclidean errors of estimated translation vector t∗ and yaw angle β∗ 
2. If you don't  include differentiable PnP in the code, how does the gradient of object pose be backprop in network ? The gradient of translation could not be propagated to the network in this code base?
Thanks.
